api/v1beta1/foundationdbcluster_types.go

```diff
@@ -470,16 +470,45 @@ type ProcessGroupStatus struct {
 	// Addresses represents the list of addresses the process group has been known to have.
 	Addresses []string `json:"addresses,omitempty"`
 	// Remove defines if the process group is marked for removal.
+	// Deprecated: Use RemovalTimestamp instead.
 	Remove bool `json:"remove,omitempty"`
+	// RemoveTimestamp if not empty defines when the process group was marked for removal.
+	RemovalTimestamp *metav1.Time `json:"removalTimestamp,omitempty"`
 	// Excluded defines if the process group has been fully excluded.
 	// This is only used within the reconciliation process, and should not be considered authoritative.
+	// Deprecated: Use ExclusionTimestamp instead.
 	Excluded bool `json:"excluded,omitempty"`
+	// ExcludedTimestamp defines when the process group has been fully excluded.
+	// This is only used within the reconciliation process, and should not be considered authoritative.
+	ExclusionTimestamp *metav1.Time `json:"exclusionTimestamp,omitempty"`
 	// ExclusionSkipped determines if exclusion has been skipped for a process, which will allow the process group to be removed without exclusion.
 	ExclusionSkipped bool `json:"exclusionSkipped,omitempty"`
 	// ProcessGroupConditions represents a list of degraded conditions that the process group is in.
 	ProcessGroupConditions []*ProcessGroupCondition `json:"processGroupConditions,omitempty"`
 }
 
+// IsExcluded returns if a process group is excluded
+func (processGroupStatus *ProcessGroupStatus) IsExcluded() bool {
+	return processGroupStatus.Excluded || (processGroupStatus.ExclusionTimestamp != nil && !processGroupStatus.ExclusionTimestamp.IsZero()) || processGroupStatus.ExclusionSkipped
+}
+
+// SetExclude marks a process group as excluded
+func (processGroupStatus *ProcessGroupStatus) SetExclude() {
+	processGroupStatus.Excluded = true
+	processGroupStatus.ExclusionTimestamp = &metav1.Time{Time: time.Now()}
+}
+
+// IsMarkedForRemoval returns if a process group is marked for removal
+func (processGroupStatus *ProcessGroupStatus) IsMarkedForRemoval() bool {
+	return processGroupStatus.Remove || (processGroupStatus.RemovalTimestamp != nil && !processGroupStatus.RemovalTimestamp.IsZero())
+}
+
+// SetRemove marks a process group for removal
+func (processGroupStatus *ProcessGroupStatus) SetRemove() {
```

Comment:
Maybe we should rename this to `MarkForRemoval` to better match with `IsMarkedForRemoval`.